test/e2e/e2e_test.go

```diff
@@ -0,0 +1,251 @@
+	if err != nil || info == nil || !info.IsDir() {
+		os.RemoveAll(cacheDir)
+		os.MkdirAll(cacheDir, 0755)
+		fmt.Println("Downloading and caching required executables...")
+
+		for _, tuple := range [][]string{{kubectlBinaryPath, kubectlURL, kubectlHash}, {kindBinaryPath, kindURL, kindHash}} {
+			if err := fetchExecutable(tuple[0], tuple[1]); err != nil {
+				t.Fatalf("unable to fetch executable: %+v", err)
+			}
+			if err := checkHash(tuple[0], tuple[2]); err != nil {
+				t.Fatalf("unable to verify downloaded blob: %+v", err)
+			}
+		}
+		fmt.Println("Done.")
+	}
+	os.RemoveAll(webhookImagePath)
+	runCommand(t, "podman", "save", "docker.io/intel/intel-fpga-admissionwebhook:devel", "-o", webhookImagePath)
+	runCommand(t, kindBinaryPath, "create", "cluster", "--name", clusterName, "--kubeconfig", kubeConfigPath, "--image", k8sImage)
+	fmt.Println("Waiting for all nodes to be ready...")
+	runCommand(t, kubectlBinaryPath, "wait", "--for=condition=Ready", "--all", "-l", "kubernetes.io/os=linux", "--timeout=120s", "node")
+	fmt.Println("All nodes are ready. Loading webhook image...")
+	// Load webhook image to the cluster
+	runCommand(t, kindBinaryPath, "load", "image-archive", "--name", clusterName, webhookImagePath)
+}
+
+func tearDown(t *testing.T) {
+	runCommand(t, kindBinaryPath, "delete", "cluster", "--name", clusterName)
+}
+
+func TestPreprogrammed(t *testing.T) {
+	defer tearDown(t)
+	setUp(t)
+
+	webhookDeployPath, err := locateWebhookDeployerScript()
+	if err != nil {
+		t.Fatalf("unable to locate webhook-deploy.sh")
+	}
+
+	runCommand(t, webhookDeployPath, "--kubectl", kubectlBinaryPath)
+	checkPodMutation(t, "fpga.intel.com/af-d8424dc4a4a3c413f89e433683f9040b")
+}
+
+func TestOrchestrated(t *testing.T) {
+	defer tearDown(t)
+	setUp(t)
+
+	webhookDeployPath, err := locateWebhookDeployerScript()
+	if err != nil {
+		t.Fatalf("unable to locate webhook-deploy.sh")
+	}
+
+	runCommand(t, webhookDeployPath, "--kubectl", kubectlBinaryPath, "--mode", "orchestrated")
+	checkPodMutation(t, "fpga.intel.com/region-ce48969398f05f33946d560708be108a")
+}
+
+func checkPodMutation(t *testing.T, expectedMutation string) {
+	podYamlFileName, err := getTempPodYaml()
+	if err != nil {
+		t.Fatalf("%+v", err)
+	}
+	defer os.RemoveAll(podYamlFileName)
+
+	runCommand(t, kubectlBinaryPath, "wait", "--for=condition=Available", "-A", "-l", "app=intel-fpga-webhook", "--timeout=30s", "deployment")
+	runCommand(t, kubectlBinaryPath, "apply", "-f", podYamlFileName)
+
+	var success bool
+	for i := 0; i < 10; i++ {
+		time.Sleep(1 * time.Second)
+		cmd := exec.Command(kubectlBinaryPath, "get", "pod", "sample-pod", "-o", "jsonpath='{.spec.containers[0].resources.limits}'")
+		cmd.Env = append(os.Environ(), "KUBECONFIG="+kubeConfigPath)
+		output, err := cmd.Output()
+		if err != nil {
+			t.Fatalf("unable to check sample pod: %+v", err)
+		}
+		fmt.Println(string(output))
+		if bytes.Contains(output, []byte(expectedMutation)) {
+			success = true
+			break
```

Comment:
can we just return from this function here? we'd not need 'success' variable in this case I guess.

Reply:
good point, fixed.